hack/common.sh

```diff
@@ -1066,6 +1066,19 @@ function os::build::environment::withsource() {
 
   docker start "${container}" > /dev/null
   docker logs -f "${container}"
+
+  # extract content from the image
+  if [[ -n "${OS_BUILD_ENV_PRESERVE}" ]]; then
+    local workingdir="$(docker inspect -f '{{ index . "Config" "WorkingDir" }}' "${container}")"
```

Comment:
Sorry, didn't see this before. You never want to do scoping (`local`, `export`) in the same line as anything else with a return code since the return code of the "other thing" - your sub shell expansion here and below for `parent` - will be aliased by the return code of the scoping statement, which is basically always 0.